### ISSUE TITLE
XMLHttpRequest.proxy api issues

### ISSUE THREAD

**Issue by hthetiot** [kaazing/http2-cache.js#3]
Initially, XMLHttpRequest.proxy was supporting this syntax:
> XMLHttpRequest.proxy(["./config-localhost.json"]);

with config-localhost.json content:
```
{
	"url": "http://localhost:8080/",
    "options": {
        "h2PushPath": "/event-stream",
        "transport": "ws://localhost:8081/",
        "debug": true
    }
}

```

This was asynchronous loading a proxy url config per array entry on XMLHttpRequest.proxy method by performing an XHR on [configPath1,configPath2]

I have then added synchronous and fixed multi-urls support, like bellow:

```
XMLHttpRequest.proxy([
        {
            "url": "http://localhost:8080/",
            "options": {
                "h2PushPath": "/event-stream",
                "transport": "ws://localhost:8081/",
                "debug": true
            }
        },
        {
            "url": "http://localhost:8082/",
            "options": {
                "h2PushPath": "/event-stream",
                "transport": "ws://localhost:8081/",
                "debug": true
            }
        }
    ]);
```

We would need now to support this:
> XMLHttpRequest.proxy("./config-urls.json");

with config-urls.json content:
```
[
        {
            "url": "http://localhost:8080/",
            "options": {
                "h2PushPath": "/event-stream",
                "transport": "ws://localhost:8081/",
                "debug": true
            }
        },
        {
            "url": "http://localhost:8082/",
            "options": {
                "h2PushPath": "/event-stream",
                "transport": "ws://localhost:8081/",
                "debug": true
            }
        }
]
```
OR

```
[
    // Default options
    "options": {
        "h2PushPath": "/event-stream",
        "transport": "ws://localhost:8081/",
        "debug": true,
        // Default proxified urls
        "urls" : [
          http://orign-server1:443/",
          http://orign-server2:443/",
        }
    },
    // Other entry no options
    {
        "url": "http://localhost:8080/"
    },
    // Other entry with options
    {
        "url": "http://localhost:8082/",
        // Per entry options
        "options": {
            "h2PushPath": "/event-stream",
            "transport": "ws://localhost:8081/",
            "debug": true
        }
    },
]

```

**Comment by dpwspoon** [comment]
We might consider making the ws and h2PushPath to level, and then having proxy urls underneath it:
Something like:
```json
{
                "h2PushPath": "/event-stream",
                "transport": "wss://rest-accelerator:443/",
                "debug": true
                "proxy" : {
                      "url": "http://orign-server1:443/",
                      "url": "http://orign-server2:443/",
                }
}
```

**Comment by hthetiot** [comment]
How about that

```
[
    // Default options
    "options": {
        "h2PushPath": "/event-stream",
        "transport": "ws://localhost:8081/",
        "debug": true,
        // Default proxified urls
        "urls" : [
          http://orign-server1:443/",
          http://orign-server2:443/",
        }
    },
    // Other entry no options
    {
        "url": "http://localhost:8080/"
    },
    // Other entry with options
    {
        "url": "http://localhost:8082/",
        // Per entry options
        "options": {
            "h2PushPath": "/event-stream",
            "transport": "ws://localhost:8081/",
            "debug": true
        }
    },
]
```

**Comment by hthetiot** [comment]
Fixed by #21
